atlasdb-jepsen-tests/src/test/java/com/palantir/atlasdb/jepsen/UniquenessCheckerTest.java

```diff
@@ -0,0 +1,133 @@
+/**
+ * Copyright 2016 Palantir Technologies
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.palantir.atlasdb.jepsen;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.junit.Test;
+
+import com.google.common.collect.ImmutableList;
+import com.palantir.atlasdb.jepsen.events.Event;
+import com.palantir.atlasdb.jepsen.events.ImmutableOkEvent;
+
+public class UniquenessCheckerTest {
+    private static final long ZERO_TIME = 0L;
+    private static final int PROCESS_0 = 0;
+    private static final int PROCESS_1 = 1;
+    private static final Long VALUE_A = 0L;
+    private static final Long VALUE_B = 1L;
+
+    @Test
+    public void shouldPassOnNoEvents() {
+        CheckerResult result = runUniquenessChecker();
+
+        assertThat(result.valid()).isTrue();
+        assertThat(result.errors()).isEmpty();
+    }
+
+    @Test
+    public void shouldSucceedOnUniqueValues() {
+        Event event1 = ImmutableOkEvent.builder()
+                .time(ZERO_TIME)
+                .process(PROCESS_0)
+                .value(VALUE_A)
+                .build();
+        Event event2 = ImmutableOkEvent.builder()
+                .time(ZERO_TIME)
+                .process(PROCESS_0)
+                .value(VALUE_B)
+                .build();
+
+        CheckerResult result = runUniquenessChecker(event1, event2);
+
+        assertThat(result.valid()).isTrue();
+        assertThat(result.errors()).isEmpty();
+    }
+
+    @Test
+    public void shouldRecordFourEventsIfThreeEqualValues() {
+        long time = 0;
+        Event event1 = ImmutableOkEvent.builder()
+                .time(time++)
+                .process(PROCESS_0)
+                .value(VALUE_A)
+                .build();
+        Event event2 = ImmutableOkEvent.builder()
+                .time(time++)
+                .process(PROCESS_0)
+                .value(VALUE_A)
+                .build();
+        Event event3 = ImmutableOkEvent.builder()
+                .time(time++)
+                .process(PROCESS_0)
+                .value(VALUE_A)
+                .build();
+
+        CheckerResult result = runUniquenessChecker(event1, event2, event3);
+
+        assertThat(result.valid()).isFalse();
+        assertThat(result.errors()).containsExactly(event1, event2, event2, event3);
```

Comment:
Right! It also just makes the code in the checker a little bit simpler; rather than using a set, or checking if the event is already there, etc.

While running the tests it is pretty confusing to see why a history fails; because the checkers don't output any sort of message to tell you what went wrong. I wrote this just to match the canonical Jepsen checkers, that also do this, but I think there could be value in improving the log output.